src/sentry/mail/utils.py

```diff
@@ -16,15 +14,24 @@
 }
 
 
+def set_user_option(UserOption, user, key, value, project):
+    inst, created = UserOption.objects.get_or_create(
+        user=user, project=project, key=key, defaults={"value": value}
+    )
+    if not created and inst.value != value:
+        inst.update(value=value)
+
+
 def migrate_project_to_issue_alert_targeting(project):
     if project.flags.has_issue_alerts_targeting:
         # Migration has already been run.
         return
     with transaction.atomic():
         # Determine whether this project actually has mail enabled
-        mail_enabled = any(
-            plugin for plugin in plugins.for_project(project, version=None) if plugin.slug == "mail"
```

Comment:
This would have always been false once we removed the mail plugin, which would basically break most on-prem users.  Switching to use using the setting means we can remove the mail plugin whenever and not have an impact.